README.md

```diff
@@ -289,9 +289,30 @@ There are three groups of tests.
 
 #### Unit Tests
 The unit tests are written using JUnit. The implementations are in
-`src/test/java/bio/terra/workspace/`. Unit tests derive from `common/BaseUnitTest.java`.
+`src/test/java/bio/terra/workspace/`.
 Some unit tests depend on the availability of a running Postgresql server.
 
+Every combination of `@MockBean` creates a distinct Spring application context. Each context holds several
+database connection pools: (WSM db, WSM Stairway db, and any amalgam connection pools). We have run out of
+database connections due to cached application contexts.
+
+To reduce the number of unique combinations, we have put **ALL** `@MockBean` into test base classes. That helps
+limit the unique combinations. You should **NEVER** code a naked `@MockBean` in a test. They should always be
+specified in these bases. That helps us control the number of unique combinations we have.
+
+The test base classes can be found in `src/test/java/bio/terra/workspace/common/`.
+
+The current inheritance for unit test base classes looks like this:
+- `BaseTest` - the base class for unit and connected tests
+    - `BaseUnitTestMocks` - the base set of mocks shared by all unit tests
+        - `BaseUnitTest` - enables the right test tags and profiles for unit testing
+            - `BaseUnitTestMockDataRepoService` - adds one mock for one unit test
```

Comment:
what does the add one mock for one unit test vs one mock used by several test means? 

it looks to me that the only difference between BaseUnitTestMockDataRepoService an BaseUnitTestMockGcpCloudContextService is as the name described.

Reply:
Each of those adds just one mock needed by a few tests. Most tests need those classes not mocked.
I tried to clarify the comment.